core/src/Kokkos_HostSpace.hpp

```diff
@@ -261,13 +261,27 @@ class SharedAllocationRecord<Kokkos::HostSpace, void>
       ;
   SharedAllocationRecord() = default;
 
+  // This constructor does not forward to the one without exec_space arg
+  // in order to work around https://github.com/kokkos/kokkos/issues/5258
+  // This constructor is templated so I can't just put it into the cpp file
+  // like the other constructor.
```

Comment:
Could you describe what the actual failure was? It seems like there was a compiler bug in play of some sort, but that's not clear.